cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sinks/KVTableSink.java

```diff
@@ -16,63 +16,75 @@
 
-  @Override
-  public void configure(StageConfigurer configurer) {
-    configurer.setName(getClass().getSimpleName());
-    configurer.setDescription("CDAP Key Value Table Dataset Batch Sink");
-    configurer.addProperty(new Property(NAME, "Dataset Name", true));
-    configurer.addProperty(new Property(
-      Properties.KeyValueTable.KEY_FIELD,
-      "The name of the field to use as the key. Its type must be bytes. Defaults to 'key'.", false));
-    configurer.addProperty(new Property(
-      Properties.KeyValueTable.VALUE_FIELD,
-      "The name of the field to use as the value. Its type must be bytes. Defaults to 'value'.", false));
+  private static final String NAME_DESC = "Dataset Name";
+  private static final String KEY_FIELD_DESC = "The name of the field to use as the key. Defaults to 'key'.";
+  private static final String VALUE_FIELD_DESC = "The name of the field to use as the value. Defaults to 'value'.";
+
+  /**
+   * Config class for KVTableSink
+   */
+  public static class KVTableConfig extends PluginConfig {
+    @Description(NAME_DESC)
+    private String name;
+
+    @Name(Properties.KeyValueTable.KEY_FIELD)
+    @Description(KEY_FIELD_DESC)
+    @Nullable
+    private String keyField = Properties.KeyValueTable.DEFAULT_KEY_FIELD;
+
+    @Name(Properties.KeyValueTable.VALUE_FIELD)
+    @Description(VALUE_FIELD_DESC)
+    @Nullable
+    private String valueField = Properties.KeyValueTable.DEFAULT_VALUE_FIELD;
+
+    public KVTableConfig(String name, String keyField, String valueField) {
+      this.name = name;
+      this.keyField = keyField;
+      this.valueField = valueField;
+    }
   }
 
-  @Override
-  protected String getDatasetType(ETLStage config) {
-    return KeyValueTable.class.getName();
+  private final KVTableConfig kvTableConfig;
+
+  public KVTableSink(KVTableConfig kvTableConfig) {
+    this.kvTableConfig = kvTableConfig;
   }
 
   @Override
-  public void initialize(ETLStage stageConfig) throws Exception {
-    keyField = stageConfig.getProperties().get(Properties.KeyValueTable.KEY_FIELD);
-    if (Strings.isNullOrEmpty(keyField)) {
-      keyField = "key";
-    }
-    valueField = stageConfig.getProperties().get(Properties.KeyValueTable.VALUE_FIELD);
-    if (Strings.isNullOrEmpty(valueField)) {
-      valueField = "value";
-    }
+  protected String getDatasetType(ETLStage config) {
+    return KeyValueTable.class.getName();
   }
 
   @Override
   public void transform(StructuredRecord input, Emitter<KeyValue<byte[], byte[]>> emitter) throws Exception {
-    Object key = input.get(keyField);
+    Object key = input.get(kvTableConfig.keyField);
     Preconditions.checkArgument(key != null, "Key cannot be null.");
     byte[] keyBytes = key instanceof ByteBuffer ? Bytes.toBytes((ByteBuffer) key) : (byte[]) key;
-    Object val = input.get(valueField);
+    Object val = input.get(kvTableConfig.valueField);
     byte[] valBytes = val instanceof ByteBuffer ? Bytes.toBytes((ByteBuffer) val) : (byte[]) val;
```

Comment:
hm thats correct. This can throw a NPE then and should be checked before doing Bytes.toBytes()